LocalHistory.py

```diff
@@ -225,23 +262,26 @@ def run(self, edit):
         history_files = get_history_files(file_name, history_dir)
         history_files = history_files[1:]
 
-        if not history_files:
+        if history_files:
+            filtered_files = filtered_history_files(history_files)
+        else:
             status_msg('Local History not found for "' + file_name + '".')
             return
 
         def on_done(index):
             if index is NO_SELECTION:
                 return
 
-            file = os.path.join(history_dir, history_files[index])
-            with open(file) as f:
-                data = f.read()
-                if PY2:
-                    data.decode('utf-8')
-                self.view.replace(edit, sublime.Region(0, self.view.size()), data)
-            self.view.run_command('save')
+            # send vars to the listener for the diff/replace view
+            from_file = os.path.join(history_dir, history_files[index])
+            from_file = from_file, os.path.basename(from_file)
+            to_file = self.view.file_name(), file_name
+            HistoryReplaceDiff.from_file = from_file
+            HistoryReplaceDiff.to_file = to_file
+            HistoryListener.listening = True
+            self.view.run_command('show_diff', {'from_file': from_file, 'to_file': to_file, 'replace': True})
 
```

Comment:
replace didn't work, I replaced it with a diff view where you can press ctrl+alt+enter to replace the file